File: guides/micronaut-graphql-todo/java/src/main/java/example/micronaut/ToDoRepository.java
package example.micronaut;

import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.PageableRepository;

import static io.micronaut.data.model.query.builder.sql.Dialect.POSTGRES;

@JdbcRepository(dialect = POSTGRES) // <1>
public interface ToDoRepository extends PageableRepository<ToDo, Long> { // <2>
}

File: guides/micronaut-graphql-todo/kotlin/src/main/kotlin/example/micronaut/ToDoRepository.kt
package example.micronaut

import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect.POSTGRES
import io.micronaut.data.repository.PageableRepository

@JdbcRepository(dialect = POSTGRES) // <1>
interface ToDoRepository : PageableRepository<ToDo?, Long?> // <2>
